src/file/UrlRe.re

```diff
@@ -1,2 +1,19 @@
+[@bs.deriving abstract]
+type t = {
+  host: string,
+  hash: string,
+  hostname: string,
+  href: string,
+  origin: string,
+  pathname: string,
+  port: string,
+  protocol: string,
+  search: string,
+  searchParams: URLSearchParamsRe.t,
+};
+
+[@bs.new] external make: string => t = "URL";
+[@bs.new] external makeWithBase: (string, string) => t = "URL";
```

Comment:
There's also a `toJSON` method that's missing: https://developer.mozilla.org/en-US/docs/Web/API/URL/toJSON